lib/active_model/serializer.rb

```diff
@@ -101,16 +172,17 @@ def self.adapter
       ActiveModel::Serializer::Adapter.lookup(config.adapter)
     end
 
+    # Used to cache serializer name => serializer class
+    # when looked up by Serializer.get_serializer_for.
     def self.serializers_cache
       @serializers_cache ||= ThreadSafe::Cache.new
     end
 
-    def self.digest_caller_file(caller_line)
-      serializer_file_path = caller_line[CALLER_FILE]
-      serializer_file_contents = IO.read(serializer_file_path)
-      Digest::MD5.hexdigest(serializer_file_contents)
-    end
-
+    # Find a serializer from a class and caches the lookup.
+    # Preferentially retuns:
+    #   1. class name appended with "Serializer"
+    #   2. try again with superclass, if present
+    #   3. nil
     def self.get_serializer_for(klass)
```

Comment:
Possibly out of the scope of this PR but how about renaming this method to something like `lookup_serializer_for`? It really makes my eyes bleed to have `serializer_for` calling `get_serializer_for`.